crates/iceberg/src/spec/transform.rs

```diff
@@ -398,6 +679,80 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_none_projection() -> Result<()> {
+        let name = "projected_name".to_string();
+        let preds = TestPredicates::new();
+
+        let transform = Transform::Void;
+        let result_unary = transform.project(name.clone(), &preds.unary)?;
+        assert!(result_unary.is_none());
+
+        let transform = Transform::Year;
+        let result_binary = transform.project(name.clone(), &preds.binary)?;
+        assert!(result_binary.is_none());
+
+        let transform = Transform::Month;
+        let result_set = transform.project(name.clone(), &preds.set)?;
+        assert!(result_set.is_none());
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_truncate_project() -> Result<()> {
```

Comment:
Could you port all the tests from Java: https://github.com/apache/iceberg/blob/main/api/src/test/java/org/apache/iceberg/transforms/TestTruncatesProjection.java